agent/src/main/java/io/type/pollution/agent/TypeCheckThrashEvent.java

```diff
@@ -0,0 +1,27 @@
+package io.type.pollution.agent;
+
+import jdk.jfr.Category;
+import jdk.jfr.Description;
+import jdk.jfr.Enabled;
+import jdk.jfr.Event;
+import jdk.jfr.Label;
+import jdk.jfr.Name;
+import jdk.jfr.StackTrace;
+
+@Name(TypeCheckThrashEvent.NAME)
+@Label("Type check cache thrashing")
+@Category("Type check")
+@Description("An interface type check is invalidating a previously cached type check result. If this happens often, it can lead to thrashing and contention of the cache field.")
+@StackTrace
+@Enabled(false)
+public class TypeCheckThrashEvent extends Event {
```

Comment:
The stack trace help - but it will print it till the tracing class, which is not very nice.
Instead, you can still reuse the existing `trace` regardless, because is free and cheap (is precomputed) and is "human actionable" the same, if pasted in the Idea output (as explained in the README).